controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,39 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment)
+                        .map(x -> new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return reduce(oldRanges).equals(reduce(newRanges));
+    }
+
+    private static List<AbstractMap.SimpleEntry<Double, Double>> reduce(List<AbstractMap.SimpleEntry<Double, Double>> ranges) {
+        List<AbstractMap.SimpleEntry<Double, Double>> result = new ArrayList<>();
+        double low = -1.0;
+        double high = -1.0;
+        for (AbstractMap.SimpleEntry<Double, Double> range : ranges) {
```

Comment:
even if its redundant scaling, if someone wants to do it, we should allow it. 
We should not dictate how people may want to use scaling or what their intention could be. As long as the input is legal we should allow it. 

Take the following example --> A customer may want to seal all segments at the end of the day and create new segments everyday with exact same key range. 
This may help the customer ensure that new files are created for the stream everyday in tier 2. And they may choose to run any batch processing on everyday data.